### DISCUSSION TITLE
Wildfire Data Download

### DISCUSSION THREAD

**Discussion by ellenalamont17** [earthlab-education/Earth-Analytics-2023-01-Intro#590] [Q&A06 Vector Data] [answered]
Context:
My machine is a Dell x64 based PC running Windows 10 with an inter i7 core. I am using (and have been using) the Google Chrome browser.

I am having trouble with downloading the Wildfire dataset from Short, 2022. When I initially did the assignment, the download worked smoothly, and I was also able to replicate it in my portfolio. While going back into my portfolio project to make some changes to the format, content, and layout, my wildfire download was running more then 40 minutes without a successful load (only took ~1 minute when it was working originally). 

So, I went back into the original assignment to check if it worked there, and I am having the same issue. To troubleshoot, I went back to the data source and recopied the data URL and replaced it in the code (fail). I changed the codespace machine type to 4-core and 8-core (both failed). And finally, I deleted the codespace and started a new one (failed). I don't really understand why it is no longer working as I have not changed the code since it worked initially.

**Data Citation:**
Short, Karen C. 2022. Spatial wildfire occurrence data for the United States, 1992-2020 [FPA_FOD_20221014]. 6th Edition. Fort Collins, CO: Forest Service Research Data Archive. https://doi.org/10.2737/RDS-2013-0009.6

~~~
# Download wildfire occurance data from geodatabase

fire_url = (
    "https://www.fs.usda.gov/rds/archive/products/RDS-2013-0009.6"
    "/RDS-2013-0009.6_Data_Format2_GDB.zip"
)

fire_dir = et.data.get_data(url=fire_url)   # Path to downloaded fire directory
fire_dir
~~~

**Comment by advythr** [comment]
Hi Ellen,

I don't think I have a solution but wanted to add that I am having the exact same problem with the same data download. I have this same problem when running the download code on my local machine (2021 MacBook Pro, 32 GB, Apple M1 Max) using Miniconda, the Earth Analytics dev environment, and VS Code/Jupyter Lab.

The download goes on for up to 30 minutes without a successful load for me, too. The data download worked originally for me when I completed the notebook assignment using the Codespaces environment some weeks ago. Now I am trying to use the same code & dataset to finish up my individual portfolio assignment, and the data download does not work.

**Comment by NobodyLivesHere** [comment]
As a workaround, you may want to consider uploading the .zip file to the local repository and then using the zipfile module to unzip to a path of your choosing (or do i manually) and then work the data from there. The file seems like it downloads directly from the site fine outside of python.

**Comment by eculler** [comment, marked as answer]
Looks like the problem is that the Forest Service is now requiring that you be a web browser to download data. Luckily, it's not too hard to pretend to be a web browser! The following code is pretending to be the Linux version of Firefox...I copied the rest of the download code from earthpy, roughly.

```
import io
import os
import zipfile

import requests

# Download wildfire occurance data from geodatabase
fire_url = (
    "https://www.fs.usda.gov/rds/archive/products/RDS-2013-0009.6"
    "/RDS-2013-0009.6_Data_Format2_GDB.zip"
)
user_agent = (
    'Mozilla/5.0 (X11; Linux x86_64; rv:60.0) '
    'Gecko/20100101 Firefox/81.0'
)
r = requests.get(url=fire_url, headers={'User-Agent': user_agent})

# Unzip data
data_path = os.path.join(
    et.io.HOME, et.io.DATA_NAME, 
    'earthpy-downloads', 'RDS-2013-0009.6_Data_Format2_GDB')
archive = zipfile.ZipFile(io.BytesIO(r.content))
archive.extractall(path=data_path)
```
